cpp/daal/include/data_management/data/numeric_table.h

```diff
@@ -165,6 +165,21 @@ class DAAL_EXPORT BlockDescriptor
         _nrows = nRows;
     }
 
+    /**
+     *  Sets data pointer to use for in-place calculation
+     *  \param[in] ptr      Shared pointer to the buffer
+     *  \param[in] nColumns Number of columns
+     *  \param[in] nRows    Number of rows
+     */
+    inline void setSharedPtr(const services::SharedPtr<DataType> & ptr, size_t nColumns, size_t nRows)
+    {
+        _xBuffer.reset();
+        _hostSharedPtr.reset();
+        _ptr   = ptr;
+        _ncols = nColumns;
+        _nrows = nRows;
```

Comment:
What about `_pPtr` and `_rawPtr` should we set them as it's done in `setBuffer` or `setPtr`?

Reply:
No, we shouldn't. if `_rawPtr` is `nullptr`, data are gathered directly from `_ptr`. The `_pPtr` pointer is just non-owning pointer to user's shared data - it is used alongside with `_rawPtr` as data owner. The logic of this method is pretty similar to the `setPtr()` declared above: https://github.com/oneapi-src/oneDAL/pull/899/files/ad099999a32445bc040c8b95f33fb2782e905956#diff-562903b08de687beba0ab37e5a40f5feR159

Reply:
Shouldn't we set it to `NULL` in this method?

Reply:
They are set to `NULL` in the constructor and in the `reset()` method that must be called after the block must be reset. In my opinion, we shouldn't reset them.